inc/oapv.h

```diff
@@ -37,7 +37,11 @@ extern "C"
 {
 #endif
 
+#ifdef OAPV_STATIC_DEFINE
+#define OAPV_EXPORT
+#else
 #include <oapv/oapv_exports.h>
+#endif
```

Comment:
I guess that the many cases use "static build" for testing, rather than dynamic link.
My suggestion is use "OAPV_DYNAMIC_EXPORT" rather than "OAPV_STATIC_DEFINE".

like this,
```
#ifdef OAPV_DYNAMIC_EXPORT
#include <oapv/oapv_exports.h>
#else
#define OAPV_EXPORT
#endif
```

I believe that CMakeLists.txt also needs to be changed if then.

Reply:
The CMake mechanism (GENERATE_EXPORT_HEADER) that generates the oapv_exports.h file expects the OAPV_STATIC_DEFINE flag for static builds.

We can add an additional flag, OAPV_DYNAMIC_EXPORT, and use it in oapv.h, but this solution goes against what CMake does and may confuse users familiar with CMake solutions.

Inside oapv_exports.h, there is conditional compilation, and for static builds, the OAPV_STATIC_DEFINE flag is expected.

Introducing the additional OAPV_DYNAMIC_EXPORT flag will make the OAPV_STATIC_DEFINE flag unnecessary, and the code fragment for static builds:
```

#  define OAPV_EXPORT
#  define OAPV_NO_EXPORT
```

will need to be moved from oapv_exports.h to oapv.h, conditioning its inclusion on the OAPV_DYNAMIC_EXPORT flag.

This will leave dead code in oapv_exports.h.

Reply:
We think it's better to keep the flag OAPV_STATIC_DEFINE

Reply:
@dkozinski 
First, I am thinking that "OAPV_STATIC_DEFINE " name is a little bit ambigous and has broad meaning.
For example, it could be a better naming If you use OAPV_STATIC_EXPORT rather than OAPV_STATIC_DEFINE.
BTW, is there any naming rule about this one in CMake? like XXX_STAIC_DEFINE ?
Otherwise, we could try to use proper name for static library building.

Second, I'm commenting to use OAPV_DYNAMIC_EXPORT instead of OAPV_STAIC_DEFINE.
oapv is a simple codec library which used to link with application, static link will be used more.
Many platforms like Android can build library without any macro defines like OAPV_STATIC_DEFINE.
And then the build will not work properly since OAPV_STATIC_DEFINE  is not defined in Android build script usually.

I don't fully understand CMake mechanism, but I'd like for you to consider the above two comments.

Reply:
1.
We can definitely change the name "OAPV_STATIC_DEFINE" to "OAPV_STATIC_EXPORT".
The parameters [BASE_NAME <base_name>] and [STATIC_DEFINE <static_define>] in the generate_export_header CMake command allow for customization of the name. 
Therefore, we can easily change "OAPV_STATIC_DEFINE" to "OAPV_STATIC_EXPORT".

However, I have ambiguous feelings about the name OAPV_STATIC_EXPORT.
I don't know if the combination of the words 'static' and 'export' is the best we can do.

When we use static libraries, export macros are generally not needed. 
Static libraries are linked at compile time, and all symbols are included in the final executable.
There is no need to control visibility in the same way as with shared libraries.

Only when we create a shared library, we need to specify which functions should be accessible to other applications or libraries that link against it. Export macros help manage this visibility.

2.
As it comes to "OAPV_DYNAMIC_EXPORT", we can provide an additional flag, OAPV_DYNAMIC_EXPORT, and use it instead CMake genetrated OAPV_STATIC_EXPORT in oapv.h file, but this solution has some small drawbacks. 

This introduces some redundancy (we've got 2 flags [OAPV_DYNAMIC_EXPORT and OAPV_STATIC_EXPORT] when one [OAPV_STATIC_EXPORT] would be sufficient).

Reply:
@dkozinski 
Ok, Let's use just only "OAPV_STATIC_DEFINE" without adding or modifying macros.